swap/src/main.rs

```diff
@@ -12,38 +12,58 @@
 )]
 #![forbid(unsafe_code)]
 
-use anyhow::{bail, Result};
+use anyhow::{bail, Context, Result};
+use cli::Options;
 use futures::{channel::mpsc, StreamExt};
 use libp2p::Multiaddr;
 use log::LevelFilter;
 use std::{io, io::Write, process};
 use structopt::StructOpt;
+use swap::{alice, bitcoin::Wallet, bob, Cmd, Rsp, SwapAmounts};
 use tracing::info;
 use url::Url;
+use xmr_btc::bitcoin::{BroadcastSignedTransaction, BuildTxLockPsbt, SignTxLock};
 
 mod cli;
 mod trace;
 
-use cli::Options;
-use swap::{alice, bitcoin::Wallet, bob, Cmd, Rsp, SwapAmounts};
-use xmr_btc::bitcoin::{BroadcastSignedTransaction, BuildTxLockPsbt, SignTxLock};
-
 // TODO: Add root seed file instead of generating new seed each run.
 // TODO: Remove all instances of the todo! macro
 
 // TODO: Add a config file with these in it.
 // Alice's address and port until we have a config file.
 pub const PORT: u16 = 9876; // Arbitrarily chosen.
 pub const ADDR: &str = "127.0.0.1";
-pub const BITCOIND_JSON_RPC_URL: &str = "127.0.0.1:8332";
+pub const BITCOIND_JSON_RPC_URL: &str = "http://127.0.0.1:8332";
+
+#[cfg(feature = "tor")]
+use swap::tor::{AuthenticatedConnection, UnauthenticatedConnection};
+#[cfg(feature = "tor")]
+use torut::onion::TorSecretKeyV3;
```

Comment:
:wrench: For feature-flagged imports I prefer fully-qualified names as opposed to use statements. It should be a bit more concise and less noisy.